src/librustc/plugin/load.rs

```diff
@@ -110,7 +107,7 @@ impl<'a> PluginLoader<'a> {
                         symbol: String) -> PluginRegistrarFun {
         // Make sure the path contains a / or the linker will search for it.
         let path = env::current_dir().unwrap().join(&path);
-        let path = old_path::Path::new(path.to_str().unwrap());
+        let path = path::Path::new(path.to_str().unwrap());
```

Comment:
You should be able to actually omit this line entirely now that we don't have to convert between old and new paths